src/base/bittorrent/private/sequentialdownloadsguard.cpp

```diff
@@ -0,0 +1,128 @@
+#include "sequentialdownloadsguard.h"
+
+#include <QDebug>
+
+#include "base/bittorrent/session.h"
+#include "base/bittorrent/torrenthandle.h"
+#include "base/logger.h"
+#include "base/utils/fs.h"
+#include "base/utils/misc.h"
+
+using namespace BitTorrent;
+
+/**
+   @brief SequentialDownloadsGuard explicit Constructor.
+ */
+SequentialDownloadsGuard::SequentialDownloadsGuard(Session *const session, QTimer *const sequentialGuardTimer)
```

Comment:
This class is superfluous, as well as the presence of "yet another" timer. All you need can be done in the TorrentHandle class (in some method called from TorrentHandle::updateStatus()).

Reply:
I know, I was deciding whether or not to do this by a timer or by react to libtorrent alert, I have decided to use timer, because of performance, so the seq. checker will be called only few times per minute for every torrent.
I did fast lookup for alerts now and it looks like __stat__ alert looks good, but it is called every second for every torrent, we don't need to check so frequently.
I investigate other alerts and let know if there is some better alert for this purpose.
We need to check only every 30 sec or so.
@arvidn Do you know from memory some better alert for this purpose?

Reply:
@silverqx, the discussed alert is best way to do it. You can divide frequency if you want. E.g.:
```c++
if ((++counter % N) == 0) {
    counter = 0;
    doNeededAction();
}
```